File: modulo1/src/ros2_ws/src/fulgor_py_examples/fulgor_py_examples/add_two_ints_server.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.srv import AddTwoInts

class AddTwoInitServerNode(Node):
    def __init__(self):
        super().__init__("add_two_ints_server")
        #el servidor de servicios se crea en el constructor, donde:
        #el 1er arg es el tipo de msj, el 2do el nombre del servidor que querramos darle,
        #el 3er arg es el cb. El nombre del 2do arg es recomendable que empiece con un verb/acción.
        self.server_ = self.create_service(AddTwoInts, "add_two_ints", self.callback_add_two_ints)
        self.get_logger().info("Servidor: Add two ints creado!")

    #callback_nombre_servicio():
    def callback_add_two_ints(self, request: AddTwoInts.Request, response: AddTwoInts.Response):
        response.sum = request.a + request.b
        self.get_logger().info(f"{request.a} + {request.b} = {response.sum}")
        #hay que hacer el return si o si, por más que ya completamos el response
        return response
    
def main (args=None):
    rclpy.init(args=args)
    node = AddTwoInitServerNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()

File: modulo1/src/ros2_ws/src/fulgor_cpp_examples/src/add_two_ints_server.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/srv/add_two_ints.hpp"
// Permite usar _1, _2 en std::bind sin escribir std::placeholders::_1, etc.
using namespace std::placeholders;
class AddTwoIntsServerNode: public rclcpp::Node
{
    public:
    //Constructor de la clase.
    AddTwoIntsServerNode(): Node("add_two_ints_server")
    {
        // Creación del servidor de servicio
        server_ = this->create_service<example_interfaces::srv::AddTwoInts>("add_two_ints", 
        std::bind(&AddTwoIntsServerNode::callbackAddTwoInts, this,_1, _2));
        RCLCPP_INFO(this->get_logger(),"Add two ints service creado!");
    }
    private:
        // Función callback que se ejecuta cuando un cliente llama al servicio
        void callbackAddTwoInts(const example_interfaces::srv::AddTwoInts::Request::SharedPtr request,
                                const example_interfaces::srv::AddTwoInts::Response::SharedPtr response)
        {
            response->sum = request->a + request->b;
            RCLCPP_INFO(this->get_logger(), "%d + %d = %d",
                                            (int)request->a, (int) request->b, (int)response->sum);
        }
        // Puntero al servidor del servicio
        rclcpp::Service<example_interfaces::srv::AddTwoInts>::SharedPtr server_;
};
int main (int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<AddTwoIntsServerNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}